snaps_boot/provision/hardware/pxe_utils.py

```diff
@@ -44,15 +44,67 @@ def __main(config, operation):
     bmc_dict = prov_dict.get('BMC')
     subnet_list = dhcp_dict.get('subnet')
     cpu_core_dict = prov_dict.get('CPUCORE')
-
+    pxe_server_configuration_listmap = tftp_dict.get('pxe_server_configuration')
+    buildPxeServer = None
+    ubuntuPxeServer = False
+    centosPxeServer = False
+    ubuntu_dict = {}
+    if pxe_server_configuration_listmap == None:
+       logger.error("Please define OS(ubuntu/centos) details in hosts.yaml to create pxe server")
+       exit(1)
+    for item in pxe_server_configuration_listmap:
+      for key in item:
+        if ("ubuntu" == key):
```

Comment:
Pep8, if statements really should not have parenthesis unless you need to split the statement into multiple lines